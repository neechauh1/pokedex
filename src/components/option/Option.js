import React, { useState } from 'react';
import './option.css';

const Option = ({ label, data, selectedData, setSelectedData }) => {
	const [isOptionActive, setIsOptionActive] = useState(false);
	const handleOptionClick = () => {
		setIsOptionActive(!isOptionActive);
	};

	const handleCheckboxClick = (item) => {
		const idx = selectedData?.indexOf(item);
		if (idx === -1) {
			setSelectedData([...selectedData, item]);
		} else {
			const newSelectedData = selectedData?.filter(
				(data) => data !== item
			);
			setSelectedData(newSelectedData);
		}
	};

	return (
		<div className='option-container'>
			<div className='option-label'>
				<span>{label}</span>
			</div>
			<div
				className={`option-btn ${isOptionActive ? 'white-class' : ''}`}
				onClick={handleOptionClick}
			>
				<button aria-label='optionbutton' className={`${isOptionActive ? 'white-class' : ''}`}>
					<span>{selectedData?.length ? selectedData[0] : ''}</span>{' '}
					<span className='btn-more-text'>
						{selectedData?.length > 1
							? `+ ${selectedData?.length - 1} More`
							: ''}
					</span>
				</button>
				<i className='arrow'></i>
			</div>
			{isOptionActive && data?.length ? (
				<div className='list'>
					{data?.map((item) => (
						<label className='option' key={item}>
							<input
								type='checkbox'
								checked={
									selectedData?.indexOf(item) === -1
										? false
										: true
								}
								onClick={() => handleCheckboxClick(item)}
							/>
							<span>{item}</span>
						</label>
					))}
				</div>
			) : null}
		</div>
	);
};

export default Option;
