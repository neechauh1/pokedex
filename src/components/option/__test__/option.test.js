import React from "react";
import { ReactDOM } from "react-dom";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import Option from "../Option";

it("Button Test case with More", () => {
  const tree = renderer.create(<Option title="More Button" type={"more"} />);
  expect(tree).toMatchSnapshot();
});

test("background color should be  white", () => {
  render(<Option />);

  //   const styles = getComputedStyle(element);

  expect(Option.background).toEqual("#C9DDE2");
});

it("Renders with a className equal to the variant", () => {
  const { isOptionActive } = render(<Option variant="default" />);

  expect(isOptionActive).toHaveClass("white-class");
});
