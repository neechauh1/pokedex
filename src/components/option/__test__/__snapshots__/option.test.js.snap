// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Test case with More 1`] = `
<div
  className="option-container"
>
  <div
    className="option-label"
  >
    <span />
  </div>
  <div
    className="option-btn "
    onClick={[Function]}
  >
    <button
      aria-label="optionbutton"
      className=""
    >
      <span />
       
      <span
        className="btn-more-text"
      />
    </button>
    <i
      className="arrow"
    />
  </div>
</div>
`;
